src/code.c

```diff
@@ -2257,12 +2257,7 @@ void CodeAssLVar (
     Expr                rhsx;           /* right hand side expression      */
 
     /* allocate the assignment                                             */
-    if ( lvar <= 16 ) {
-        ass = NewStat( T_ASS_LVAR + lvar, 2 * sizeof(Stat) );
-    }
-    else {
-        ass = NewStat( T_ASS_LVAR,        2 * sizeof(Stat) );
-    }
```

Comment:
I ran the following snippets (repeatedly, compared average running times)

```
G := Group(AtlasGenerators("Fi24", 1).generators);;
gap> t := NanosecondsSinceEpoch();; Size(G);; t := NanosecondsSinceEpoch() - t;
```

```
G := Group(AtlasGenerators("Fi23", 1).generators);;
gap> t := NanosecondsSinceEpoch();; Size(G);; t := NanosecondsSinceEpoch() - t;
```

```
f := function()
    local t,i,
          r0,r1,r2,r3,r4,r5,r6,r7,
          r8,r9,r10,r11,r12,r13,r14,r15,
          r16,r17,r18,r19,r20,r21,r22,r23,r24;

    t := NanosecondsSinceEpoch();
    for i in [1..1000000] do
        r0 := Random([1..100]);
        r1 := r0 + r0;
    od;
    t := NanosecondsSinceEpoch() - t;
    Print("t1: ", t, "\n");

    t := NanosecondsSinceEpoch();
    for i in [1..1000000] do
        r24 := Random([1..100]);
        r23 := r24 + r24;
    od;
    t := NanosecondsSinceEpoch() - t;
    Print("t2: ", t, "\n");
end;
f();
```

Without any measurable performance impact. If anyone has any more suggestions for benchmarks, I am all ears, also feel free to run this on your machine to cross-check.